api_app/analyzers_manager/observable_analyzers/greynoise_labs.py

```diff
@@ -0,0 +1,65 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+import requests
+
+from api_app.analyzers_manager.classes import ObservableAnalyzer
+from api_app.analyzers_manager.exceptions import AnalyzerRunException
+from tests.mock_utils import MockUpResponse, if_mock_connections, patch
+
+queries = {
+    "noiserank": {
+        "query_string": "query NoiseRank($ip: String) { noiseRank(ip: $ip) \
+            { queryInfo { resultsAvailable resultsLimit } ips { ip noise_score \
+                sensor_pervasiveness country_pervasiveness payload_diversity \
+                    port_diversity request_rate } } }",
+        "ip_required": True,
+    },
+    "topknocks": {
+        "query_string": "query TopKnocks($ip: String) { topKnocks(ip: $ip) \
+            { queryInfo { resultsAvailable resultsLimit } knock { last_crawled \
+            last_seen source_ip knock_port title favicon_mmh3_32 \
+            favicon_mmh3_128 jarm ips emails links tor_exit headers apps } } } ",
+        "ip_required": True,
+    },
+    "topc2s": {
+        "query_string": "query TopC2s { topC2s { queryInfo \
+        { resultsAvailable resultsLimit } c2s { source_ip c2_ips \
+        c2_domains payload hits pervasiveness } } } "
+    },
+}
+
```

Comment:
I think the analyzer is cool. The only problem regarding this analyzer is this type of query that does not support IP addresses anymore.

For this cases, we usually make the analyzer work in a different way. We maintain a local cache of the data extracted from the Greynoise endpoint (a file in the system) and we open it once the analyzer is triggered.

Thanks to the `update` method, it is possible to define how to update this file and when.
Please check other analyzers like `Tor`, `Maxmind`, `Feodo Tracker`, etc that do something very similar to what I mentioned

Reply:
Hey @mlodic , I have added the `update` method for this class in a new commit.